README.md

```diff
@@ -122,34 +182,100 @@ yum install mosquitto
 
 See [mosquitto official website](https://mosquitto.org/download/) for more information.
 
-### Build Edge
+KubeEdge has certificate based authentication/authorization between cloud and edge. Certificates can be generated using openssl. Please follow below steps is any other method of generation of certificates isn't available.
+
+#### Install openssl
+
+If openssl is not already present using below command to install openssl
+
+```shell
+apt-get install openssl
+```
+#### Generate Certificates
+
+RootCA certificate and a cert/key pair is required to have a setup for KubeEdge. Same cert/key pair can be used in both cloud and edge. 
+```shell
+# Generete Root Key
+openssl genrsa -des3 -out rootCA.key 4096
+# Generate Root Certificate
+openssl req -x509 -new -nodes -key rootCA.key -sha256 -days 1024 -out rootCA.crt
+# Generate Key
+openssl genrsa -out kubeedge.key 2048
+# Generate csr, Fill required details after running the command
+openssl req -new -key kubeedge.key -out kubeedge.csr
+# Generate Certificate
+openssl x509 -req -in kubeedge.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out kubeedge.crt -days 500 -sha256 
+```
+
+### Clone KubeEdge
 
 Clone KubeEdge
 
 ```shell
 git clone https://github.com/kubeedge/kubeedge.git $GOPATH/src/github.com/kubeedge/kubeedge
+cd $GOPATH/src/github.com/kubeedge/kubeedge
+```
+
+### Build Cloud
+
+```shell
+cd $GOPATH/src/github.com/kubeedge/kubeedge/cloud/edgecontroller
+make # or `make edgecontroller`
+```
+
+### Build Edge
+
+```shell
 cd $GOPATH/src/github.com/kubeedge/kubeedge/edge
-make # or `make edge_core`
+make # or `make edgecontroller`
 ```
+
 KubeEdge can also be cross compiled to run on ARM based processors.
 Please click [Cross Compilation](docs/setup/cross-compilation.md) for the instructions.
 
+## Run KubeEdge
+
+### Run Cloud
+
+```shell
+cd $GOPATH/src/github.com/kubeedge/kubeedge/cloud/edgecontroller
+# run edge controller
+# `conf/` should be in the same directory as the binary
+# verify the configurations before running cloud(edgecontroller)
+./edgecontroller
+```
+
 ### Run Edge
 
+We have provided a sample node.json to add a node in kubernetes. Please make sure edge-node is added in kubernetes. Run below steps to add edge-node
+  
+```shell
+kubectl apply -f $GOPATH/src/github.com/kubeedge/kubeedge/build/node.json
+```
+
+Run Edge
+
 ```shell
 # run mosquitto
 mosquitto -d -p 1883
 
 # run edge_core
 # `conf/` should be in the same directory as the binary
+# verify the configurations before running edge(edge_core)
 ./edge_core
 # or
 nohup ./edge_core > edge_core.log 2>&1 &
 ```
 
 If you are using HuaweiCloud IEF, then the edge node you created should be running (check it in the IEF console page).
 
+### Deploy Application
 
+Try out a sample application deployment by following below steps
```

Comment:
We don't need to give details of how to create deployments etc, maybe we can give references to Kubernetes documentation on how to create deployments, secrets, config maps.

Reply:
Just a sample tested implementation for now. We have a few constraints like hostPort == containerPort. We will get a document is place soon for all supported configs after which we will delete this file.